trade-finance-manager-api/src/v1/services/deal-cancellation/deal-cancellation.service.ts

```diff
@@ -0,0 +1,64 @@
+import { endOfDay, format, isAfter, toDate } from 'date-fns';
+import { DATE_FORMATS, DEAL_TYPE, TEAMS, TfmDealCancellation } from '@ukef/dtfs2-common';
+import sendTfmEmail from '../send-tfm-email';
+import { CANCEL_DEAL_PAST_DATE, CANCEL_DEAL_FUTURE_DATE } from '../../../constants/email-template-ids';
+import * as api from '../../api';
+import { formatFacilityIds } from './helpers/format-facility-ids';
+import { UKEF_ID } from '../../../constants/deals';
+
+export class DealCancellationService {
+  /**
+   * Sends email to PIM email address confirming the cancellation has been submitted
+   * @param ukefDealId ukef deal id
+   * @param dealCancellation deal cancellation object
+   * @param facilityIds ukef facility ids
+   * @private - This function is exported for unit testing only and should not be used outside of the service
+   */
+  public static async sendDealCancellationEmail(ukefDealId: string, dealCancellation: TfmDealCancellation, facilityIds: string[]): Promise<void> {
+    const effectiveFromDate = toDate(dealCancellation.effectiveFrom);
+    const endOfToday = endOfDay(new Date());
+
+    const { email: pimEmail } = await api.findOneTeam(TEAMS.PIM.id);
+
+    if (!pimEmail) {
+      throw Error('Failed to send deal cancellation email to PIM');
+    }
+
+    const emailTemplateId = isAfter(effectiveFromDate, endOfToday) ? CANCEL_DEAL_FUTURE_DATE : CANCEL_DEAL_PAST_DATE;
+
+    const formattedEffectiveFromDate = format(dealCancellation.effectiveFrom, DATE_FORMATS.D_MMMM_YYYY);
+    const formattedBankRequestDate = format(dealCancellation.bankRequestDate, DATE_FORMATS.D_MMMM_YYYY);
+
+    await sendTfmEmail(emailTemplateId, pimEmail, {
+      ukefDealId,
+      effectiveFromDate: formattedEffectiveFromDate,
+      bankRequestDate: formattedBankRequestDate,
+      cancelReason: dealCancellation.reason || '-',
```

Comment:
Add this above please.

```ts
const cancelReason = dealCancellation?.reason ?? '-'
```


```suggestion
      cancelReason,
```

Reply:
We need to use `||` since we want it to default to `'-'` when the reason is an empty string. The dealCancellation is definitely defined here too (using zod payload validation) so I dont think using optional chaining is necessary